common/lib/util/multicaster.ts

```diff
@@ -0,0 +1,49 @@
+import Logger from './logger';
+
+type AnyFunction = (...args: any[]) => unknown;
+
+export interface MulticasterInstance extends Function {
+	(...args: unknown[]): void;
+	push: (fn: AnyFunction) => void;
+}
+
+class Multicaster {
+	members: Array<AnyFunction>;
+
+  // Private constructor; use static Multicaster.create instead
+	private constructor(members?: Array<AnyFunction | undefined>) {
+		this.members = members as Array<AnyFunction> || [];
+	}
+
+	call(...args: unknown[]): void {
+		for (const member of this.members) {
+			if (member) {
+				try {
+          member(...args);
```

Comment:
I'm trying to ignore the indentation issue here as I know you plan to fix this later with [Prettier](https://github.com/ably/ably-js/issues/684) (/ [ESLint](https://github.com/ably/ably-js/issues/685)), but I couldn't simply pass by without commenting as it does make it harder to review when things like this stand out (cognitive load on reviewers as they parse the code for themselves).